Tasks/Common/packaging-common/provenance.ts

```diff
@@ -0,0 +1,163 @@
+        let buildData = {
+            "System.CollectionId": tl.getVariable("System.CollectionId"),
+            "System.TeamProjectId": tl.getVariable("System.TeamProjectId"),
+            "Build.BuildId": buildId,
+            "Build.BuildNumber": tl.getVariable("Build.BuildNumber"),
+            "Build.DefinitionName": tl.getVariable("Build.DefinitionName"),
+            "Build.Repository.Name": tl.getVariable("Build.Repository.Name"),
+            "Build.Repository.Provider": tl.getVariable("Build.Repository.Provider"),
+            "Build.Repository.Id": tl.getVariable("Build.Repository.Id"),
+            "Build.SourceBranch": tl.getVariable("Build.SourceBranch"),
+            "Build.SourceBranchName": tl.getVariable("Build.SourceBranchName"),
+            "Build.SourceVersion": tl.getVariable("Build.SourceVersion")
+        }
+
+        var sessionRequest: SessionRequest = { 
+            feed: feedId,
+            source: "InternalBuild",
+            data: buildData
+        }
+
+        return sessionRequest;
+    }
+}
+
+export class ProvenanceApi extends vstsClientBases.ClientApiBase {
+    constructor(baseUrl: string, handlers: VsoBaseInterfaces.IRequestHandler[], options?: VsoBaseInterfaces.IRequestOptions) {
+        super(baseUrl, handlers, "node-packageprovenance-api", options);
+    }
+
+    /**
+     * Creates a session, a wrapper around a feed that can store additional metadata on the packages published to the session.
+     * 
+     * @param {SessionRequest} sessionRequest - The feed and metadata for the session
+     * @param {string} protocol - The protocol that the session will target
+     */
+    public async createSession(
+        sessionRequest: SessionRequest,
+        protocol: string
+        ): Promise<SessionResponse> {
+
+        return new Promise<SessionResponse>(async (resolve, reject) => {
+
+            let routeValues: any = {
+                protocol: protocol
+            };
+
+            try {
+                let verData: ClientVersioningData = await this.vsoClient.getVersioningData(
+                    "5.0-preview.1",
+                    "Provenance",
+                    "503B4E54-EBF4-4D04-8EEE-21C00823C2AC",
+                    routeValues);
+
+                let url: string = verData.requestUrl;
+
+                let options: restclient.IRequestOptions = this.createRequestOptions('application/json', 
+                                                                                verData.apiVersion);
+
+                let res: restclient.IRestResponse<SessionResponse>;
+                res = await this.rest.create<SessionResponse>(url, sessionRequest, options);
+                let ret = this.formatResponse(res.result,
+                                              null,
+                                              false);
+
+                resolve(ret);
+            }
+            catch (err) {
+                reject(err);
```

Comment:
You can add logs here in debug mode.

Reply:
Decided not to add it here, since this is generated code that will eventually be moved to azure-devops-node-api. The method that calls this does log the error. I'll make the API class private for now.